src/main/scala/abalone/variant/Variant.scala

```diff
@@ -90,18 +101,82 @@ abstract class Variant private[variant] (
     }
   }
 
-  def validMovesOf1(situation: Situation): Map[Pos, List[Move]] =
-    situation.board.piecesOf(situation.player).flatMap {
+  // @TODO: move this into the validMoves method and make it become private as it's used by validMovesOf1 AND validMovesOf2
+  def turnPieces(s: Situation): PieceMap = s.board.piecesOf(s.player)
+
+  def validMovesOf1(s: Situation): Map[Pos, List[Move]] =
+    this.turnPieces(s).flatMap {
       case ((pos, piece)) =>
         Map(pos ->
           pos.neighbours.flatten
-          .filterNot(situation.board.pieces.contains(_))
+          .filterNot(s.board.pieces.contains(_))
           .map(landingSquare => 
-            Move(piece, pos, landingSquare, situation, boardAfter(situation, pos, landingSquare), false)
+            Move(piece, pos, landingSquare, s, boardAfter(s, pos, landingSquare), false)
           )
         )
     }.toMap
 
+  def validMovesOf2(s: Situation): Map[Pos, List[(String, Move)]] = {
+    def generateMove(orig: Pos, dest: Pos, category: String) = category match {
```

Comment:
just a comment for future, given the cases are very similar (and you only have two) you could remove and have an if statement inside the last input of Move (capture) which would shorten the code. But its fine the way you have written it.